model/task/db.go

```diff
@@ -985,6 +1015,29 @@ const VersionLimit = 50
 // It attempts to return the most recent display name for each build variant to avoid returning duplicates caused by display names changing.
 // It only checks the last 50 versions that ran for a given task name.
 func FindUniqueBuildVariantNamesByTask(projectId string, taskName string, repoOrderNumber int) ([]*BuildVariantTuple, error) {
+	noLookupPipeline := variantByTaskPipeline(projectId, taskName, repoOrderNumber)
+	lookupPipeline := variantByTaskPipelineWithLookup(projectId, taskName, repoOrderNumber)
+	facet := bson.M{
+		"$facet": bson.M{
+			"noLookup": noLookupPipeline,
+			"lookup":   lookupPipeline,
+		},
+	}
+	result := []*buildVariantTupleResult{}
+	if err := Aggregate([]bson.M{facet, {"$project": bson.M{
+		"build_variants": bson.M{
+			"$setUnion": []string{"$lookup", "$noLookup"},
```

Comment:
can we rename these variables. Its confusing seeing `$lookup` as a parameter since it represents a mongodb aggregation step.